### PR TITLE
Eslint config project improvements

### PR DESCRIPTION
**Pull request by lb-**
**Do not merge this PR yet - discussion needed**

The goal of this PR is to start to get this project updated to be easier to develop with, easier to update soon and to align with the new [stylelint-config-wagtail](https://github.com/wagtail/stylelint-config-wagtail) project.

Discussion/WIP:
* Are we happy to go in this direction
* Tests need to be added (if practical, it might be easier to upgrade eslint first)
* What version of eslint to align with, v6.0.0 alpha just came out, might be worth getting on this version as it progresses in the coming weeks

Improvements:
* Add .editorconfig, prettier (formatting), jest (testing - no tests yet)
* Add License
* Fleshed out readme (with dev instructions)

Related Issues:
* https://github.com/wagtail/wagtail/issues/5252
* https://github.com/wagtail/wagtail/issues/5246

### CODE REVIEW

**Comment by lb-** [comment]
There is an extension of this PR here:

https://github.com/wagtail/eslint-config-wagtail/compare/master...lb-:eslint-upgrade?expand=1

* The tests are not fully working as yet but close
* Added ONE extends (eslint:recommended) and reviewed all other rules, no changes needed
* Remove two rules that are already part of eslint:recommended

Would love feedback on this, if we are good to go down this path.

*Important*: we should probably have a base and full config (similar to airbnb) as some projects will not use the react part, this has not been done yet.

**Comment by lb-** [comment]
@thibaudcolas  + @jonnyscholes  + @nmorduch 

Would love any feedback on this, or mention anyone that should also be in the loop

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+charset = utf-8
+
+[*.{css,py,scss,rst}]
+indent_size = 4
+
+[*.{js,json,yml}]
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (thibaudcolas) [inline .editorconfig]: Missing newline

```suggestion
trim_trailing_whitespace = false

```

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+10
```
Comment (thibaudcolas) [inline .nvmrc]: Missing newline

```suggestion
10

```

File: package.json [hunk 3]
```diff
@@ -5,24 +5,41 @@
...
   "repository": {
     "type": "git",
     "url": "git+https://github.com/wagtail/eslint-config-wagtail.git"
   },
+  "dependencies": {
+    "eslint-config-airbnb": "^9.0.1",
+  },
+  "devDependencies": {
+    "eslint": "^2.12.0",
+    "jest": "24.7.1",
```
Comment (thibaudcolas) [inline package.json]: Why pin to a specific version?

```suggestion
    "jest": "^24.7.1",
```

**Review by thibaudcolas — APPROVED** [review]
I know you prefaced your PR with "discussion needed"… but to me all your changes are clearly big positives, so I’ll merge it anyway. There are a few things I’d have suggested to do differently but really nothing significant. Let’s have a look at your https://github.com/wagtail/eslint-config-wagtail/compare/master...lb-:eslint-upgrade?expand=1 next!

**Comment by lb-** [comment]
@thibaudcolas  - awesome, thanks heaps.
Sorry for the delay in replying but thanks for the feedback and merging this one in.
I will attempt to get my head around this again soon and see if I can prep another PR.

**Pull request merged**